pkg/rekor/v2/verifier.go

```diff
@@ -0,0 +1,180 @@
+	}
+
+	verifier, err := signature.LoadVerifier(matchingLogInstance.PublicKey, matchingLogInstance.HashFunc)
+	if err != nil {
+		return nil, err
+	}
+	return verifier, nil
+}
+
+func RunConsistencyCheck(ctx context.Context, rekorShards map[string]ShardInfo, activeShardOrigin string, logInfoFile string) (*log.Checkpoint, error) {
+	// First, we select the correct shard. Most of the time this will be
+	// the latest active shard (with origin == activeShardOrigin), but
+	// in situations where the previously stored checkpoint is from an older
+	// shard, we have to:
+	// - Verify the previous checkpoint against the last checkpoint of the older shard
+	// - Store the last checkpoint of the *newest* shard as the last-seen checkpoint
+	// in order to migrate from the old shald to the new one.
+
+	// Fetch (and verify) the latest checkpoint of the latest active shard
+	// This is the checkpoint that will be saved to `logInfoFile`.
+	latestActiveCheckpoint, _, err := (*rekorShards[activeShardOrigin].client).ReadCheckpoint(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get current checkpoint: %v", err)
+	}
+
+	var prevCheckpoint *log.Checkpoint
+	fi, err := os.Stat(logInfoFile)
+	if err == nil && fi.Size() != 0 {
+		// Read the latest saved checkpoint from the log file
+		prevCheckpoint, err = file.ReadLatestCheckpointRekorV2(logInfoFile)
+		if err != nil {
+			return nil, fmt.Errorf("reading checkpoint log: %v", err)
+		}
+
+		// The new checkpoint we fetch for the consistency check has to be from the same
+		// shard as the previous checkpoint.
+		rekorClient := *rekorShards[activeShardOrigin].client
+		newCheckpoint := latestActiveCheckpoint
+		if prevCheckpoint.Origin != activeShardOrigin {
+			rekorClient = *rekorShards[prevCheckpoint.Origin].client
+			newCheckpoint, _, err = rekorClient.ReadCheckpoint(ctx)
+			if err != nil {
+				return nil, fmt.Errorf("failed to get current checkpoint: %v", err)
+			}
+		}
+
+		// Build the consistency proof between the tree sizes of the previous (stored)
+		// checkpoint and the newest fetched checkpoint
+		pb, err := tclient.NewProofBuilder(ctx, newCheckpoint.Size, rekorClient.ReadTile)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get proof builder: %v", err)
+		}
+		consistencyProof, err := pb.ConsistencyProof(ctx, prevCheckpoint.Size, newCheckpoint.Size)
+		if err != nil {
+			return nil, fmt.Errorf("failed to build consistency proof: %v", err)
+		}
+
+		err = proof.VerifyConsistency(rfc6962.DefaultHasher, prevCheckpoint.Size, newCheckpoint.Size, consistencyProof, prevCheckpoint.Hash, newCheckpoint.Hash)
+		if err != nil {
+			return nil, fmt.Errorf("consistency check failed: %v", err)
+		}
+
+		fmt.Fprintf(os.Stderr, "Root hash consistency verified - Current Size: %d Root Hash: %s - Previous Size: %d Root Hash %s\n",
+			newCheckpoint.Size, hex.EncodeToString(newCheckpoint.Hash), prevCheckpoint.Size, hex.EncodeToString(prevCheckpoint.Hash))
```

Comment:
i'd make this a separate function like what's done in v1 with `verifyCheckpointConsistency`.

Reply:
not sure I agree. This function is already called `RunConsistencyCheck`, so it's clear it's about checking the consistency, and it's not too big (~70 lines long). 

Moving the meat of it to a separate `verifyCheckpointConsistency` does not add much in terms of readability IMHO.